src/main/java/com/example/demo/test/Controllers/MachineController.java

```diff
@@ -0,0 +1,69 @@
+package com.example.demo.test.Controllers;
+import com.example.demo.test.Models.Machine;
+import com.example.demo.test.Services.JobService;
+import com.example.demo.test.Services.MachineService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import java.util.Optional;
+@RestController
+@RequestMapping("/Machine")
+public class MachineController
+{
+    @Autowired
+    MachineService machineService;
+    @Autowired
+    JobService jobService;
+    @Autowired
+    ObjectMapper mapper;
+
+    public MachineController()
+    {
+    }
+    @PostMapping("/create")
+    public Machine createMachine(@RequestBody Machine machine1)
+    {
+        if (machineService.findMachine(machine1.getMachineId()).isPresent())
```

Comment:
there is no need to test the machineId as it will not be taken from the body as it is auto generated, so delete this block